### PR TITLE
fix: BrowserTestRunner called incorrect super methods

### PR DESCRIPTION
**Pull request by samthor**
### Description

This is a tiny fix; the browser `VitestRunner` impl wasn't referencing any types, so some of its `super` calls were invalid.

Adds constructor types and fixes the calls.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — APPROVED** [review]
Thank you 🙏

**Comment by samthor** [comment]
looks like the CI is a bit borked re: #4628 (since that is a docs only change). Also, I guess PR authors can't merge.

File: packages/browser/src/client/runner.ts [hunk 1]
```diff
@@ -39,8 +42,8 @@ export function createBrowserRunner(original: any, coverageModule: CoverageHandl
       }
     }
 
-    async onAfterRunFiles() {
-      await super.onAfterRun?.()
+    async onAfterRunFiles(files: File[]) {
+      await super.onAfterRunFiles?.(files)
```
Comment (AriPerkkio) [inline packages/browser/src/client/runner.ts:46]: I remember there being some changes done here earlier:

- https://github.com/vitest-dev/vitest/issues/3514
- https://github.com/vitest-dev/vitest/pull/3806

I'll need to re-check this later today.

Comment (AriPerkkio) [reply packages/browser/src/client/runner.ts:46]: Looks good!

**Pull request merged**